plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XIncrementalBuilder.java

```diff
@@ -279,7 +279,7 @@ protected void unloadResource(URI uri) {
 						operationCanceledManager.checkCanceled(cancelIndicator);
 						// trigger init
 						resource.getContents();
-						EcoreUtil2.resolveLazyCrossReferences(resource, cancelIndicator);
+						EcoreUtil2.resolveLazyCrossReferences(resource, CancelIndicator.NullImpl);
```

Comment:
Normally, resolving the cross-references can be a (very) long-running operation. Why is it ok to ignore the cancel indicator here by passing in `NullImpl`?

(similarly in line 293 below)

Reply:
Thanks for the quick response!

In that case it would be good to have a TODO in this line to not forget to change it back once the LSP CancelIndicator is fixed. Or we could add a comment to some task to check all references to `CancelIndicator.NullImpl`.